test/test_rosdep_rospkg_loader.py

```diff
@@ -179,3 +179,8 @@ def test_RosPkgLoader_get_loadable():
     keys = loader.get_loadable_views()
     for s in ['ros', 'empty', 'invalid', 'stack1']:
         assert s in keys
+
+
+def test_rospkg_loader_all_dep_types():
+    from rosdep2.rospkg_loader import all_dep_types
+    assert all_dep_types() == {'build', 'buildtool', 'build_export', 'buildtool_export', 'exec', 'test', 'doc'}, all_dep_types()
```

Comment:
This is a test file so fixed values may make more sense, but I still want to raise this thoughts: Should the types be taken from the upstream `catkin_pkg.package.Package` like done [in the source](https://github.com/ros-infrastructure/rosdep/pull/789/files#diff-68811be333f5db97a55f12cb74a4a059af80d63e4d599c7a51ad104d647d6c93R62)?

Reply:
I think this strategy isn't adequate because it would use the same principle the function being tested uses as if I was doing `asssert all_dep_types() == all_dep_types()`, which would be always true. Imagine if a new dependency type is added to `Package` and we don't want to consider it just yet. The tests would pass without our intention.

On the other hand, you might argue that this list could be constant in the source code and the test could check the list against all the dep types in `Package`. I think this is valid too and maybe it is in better agreement with my [dissatisfaction](https://github.com/ros-infrastructure/rosdep/pull/789#discussion_r540844183) with this function. I think the maintainers could help us with this decision.

Reply:
> On the other hand, you might argue that this list could be constant in the source code and the test could check the list against all the dep types in `Package`.

+1 for that (I'm not a maintainer).